server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1273,7 +1273,13 @@ private void parseSource(DefaultSearchContext context, SearchSourceBuilder sourc
         if (source.trackTotalHitsUpTo() != null
             && source.trackTotalHitsUpTo() != SearchContext.TRACK_TOTAL_HITS_ACCURATE
             && context.scrollContext() != null) {
-            throw new SearchException(shardTarget, "disabling [track_total_hits] is not allowed in a scroll context");
+            // this check is now performed in the coord node, data nodes still need to check it for bw comp.
+            // You could have a coord node in the previous version that does not perform the check, in which case the shard must check.
+            throw new SearchException(
+                shardTarget,
+                "disabling [track_total_hits] is not allowed in a scroll context",
+                new IllegalArgumentException("disabling [track_total_hits] is not allowed in a scroll context")
```

Comment:
this is hacky in that we create two exceptions. SearchException so that we get the shard target information, and IllegalArgumentException so that we get a 400 status code. I really would like to avoid introducing a new exception type, especially as these checks are here only for bw comp and can be removed in the future. Perhaps this could just be an IllegalArgumentException? All shards are going to return the same error anyways?

Reply:
I went for IllegalArgumentException, I think it's less hacky and the shard target is really not needed in this case.